dependent-service-downtime-and-ps-adaptor-behavior.md

```diff
@@ -0,0 +1,64 @@
+1. GP2GP Translator is Down
+   - Scenario: The GP2GP Translator is not operational and cannot send patient records.
+   - Expected Behavior: No messages are sent to the incumbent system while the translator is down. Once the GP2GP Translator recovers, 
+     the transfer process resumes, and messages are processed as normal.
+   - Recovery: Automatic upon Translator recovery; transfer resumes without manual intervention.
+2. Queue is Down
+   - Scenario: The queue responsible for transferring data between the GP2GP Translator and PS Adaptor is down.
+   - Expected Behavior: The initial request is never sent to the incumbent system because the queue is unavailable.
+   - Recovery: After the queue is restored, the request can be processed normally, but any messages queued during downtime are not sent.
+3. PS Adaptor Database (DB) is Down
+   - Scenario: The PS Adaptor's database is not operational.
+   - Expected Behavior: The initial request to the incumbent system is not sent because the adaptor cannot access the necessary data.
+   - Recovery: After the database is restored, the system may resume operations, but requests during the outage are not processed.
+4. Facade Application is Down
+   - Scenario: The facade application that works with the PS Adaptor is down.
+   - Expected Behavior: The initial request to the incumbent system is not sent. The communication between the PS Adaptor and the incumbent system is blocked.
+   - Recovery: Once the facade application is operational, the system can process requests as normal.
+5. MHS Adaptor is Down
+   - Scenario: The MHS Adaptor (including its database) is not operational.
+   - Expected Behavior: The EHR request is not sent to the incumbent system while the MHS Adaptor is down. 
+     Once the MHS Adaptor's database recovers, the request is sent as normal.
+   - Recovery: Automatic upon recovery of the MHS Adaptor’s database.
+6. MHS Adaptor Inbound Queue is Down
+   - Scenario: The inbound queue managed by the MHS Adaptor is down.
+   - Expected Behavior: The EHR request is successfully sent to the incumbent system.
+   - Recovery: No action is required as the request is still processed.
+7. MHS Adaptor Outbound Queue is Down
+   - Scenario: The outbound queue managed by the MHS Adaptor is down.
+   - Expected Behavior: The request is not sent to the incumbent system while the outbound queue is down. 
+     Once the outbound service recovers, the request is processed and sent.
+   - Recovery: The system automatically sends the request when the outbound queue becomes operational again.
+8. File Storage is Down
+   - Scenario: The file storage system is not operational.
+   - Expected Behavior: Despite the outage, the EHR request is still sent to the incumbent system. 
+     However, any operations that rely on file storage will fail.
+   - Recovery: File storage recovery may be necessary for non-EHR-related operations, but the EHR transfer proceeds.
```

Comment:
No, we saying that as per the test that was done in NIAD-2369 the fact that the file system was down didn't affect the service however potentially it might have some effects on the transfer especially in those cases (which might not have been included in the test) that rely on the file storage.

Reply:
Oh strange. How could any documents get saved, if the file storage is down. This test seems very unlikely?!

Documentation which is wrong is worse than no documentation.